compiler/backend/backendScript.sml

```diff
@@ -108,31 +141,39 @@ val to_data_def = Define`
 val to_word_def = Define`
   to_word c p =
   let (c,p) = to_data c p in
-  let (col,p) = data_to_word$compile c.data_conf c.word_to_word_conf c.lab_conf.asm_conf p in
+  let (col,p) = data_to_word$compile c.data_conf c.word_to_word_conf c.asm_conf p in
   let c = c with word_to_word_conf updated_by (λc. c with col_oracle := col) in
   (c,p)`;
 
 val to_stack_def = Define`
   to_stack c p =
   let (c,p) = to_word c p in
-  let (c',p) = word_to_stack$compile c.lab_conf.asm_conf p in
+  let (c',p) = word_to_stack$compile c.asm_conf p in
   let c = c with word_conf := c' in
   (c,p)`;
 
 val to_lab_def = Define`
   to_lab c p =
-  let (c,p) = to_stack c p in
-  let p = stack_to_lab$compile
-    c.stack_conf c.data_conf (2 * max_heap_limit (:'a) c.data_conf - 1)
-    (c.lab_conf.asm_conf.reg_count - (LENGTH c.lab_conf.asm_conf.avoid_regs +3))
-    (c.lab_conf.asm_conf.addr_offset) p in
-  (c,p:'a prog)`;
+  case c.late_conf of
+  | ToTarget stack_to_lab_conf lab_conf =>
+    let (c,p) = to_stack c p in
+    let p = stack_to_lab$compile
+      stack_to_lab_conf c.data_conf (2 * max_heap_limit (:'a) c.data_conf - 1)
+      (c.asm_conf.reg_count - (LENGTH c.asm_conf.avoid_regs +3))
+      (c.asm_conf.addr_offset) p in
+    SOME (c,p:'a prog)
+  | _ => NONE`;
```

Comment:
In case there is no configuration for `stack→ lab`, this function may now return `NONE`.